### PR TITLE
Items per page refactor on admin side

### PR DESCRIPTION
**Pull request by vesalink**
https://bcmines.atlassian.net/browse/EE-729

### CODE REVIEW

File: src/app/shared/components/table-template/table-template.component.ts [hunk 1]
```diff
@@ -26,6 +31,13 @@ export class TableTemplateComponent implements OnInit, OnChanges, OnDestroy {
 
   ngOnInit() {
     this.loadComponent();
+    this.activePageSize = parseInt(this.data.paginationData.pageSize, 10);
+    const pageSizeTemp = [10, 25, 50, 100, parseInt(this.data.paginationData.totalListItems, 10)];
```
Comment (mtCarto) [inline src/app/shared/components/table-template/table-template.component.ts]: Curious why the list has 10 as the last element as well as the first?

**Comment by mtCarto** [comment]
Looks pretty good, noticed one bug we should address. When on a page other than 1, clicking the "Show All" button takes you back to general search home page, instead of listing all projects.

**Comment by vesalink** [comment]
> Looks pretty good, noticed one bug we should address. When on a page other than 1, clicking the "Show All" button takes you back to general search home page, instead of listing all projects.

pr updated addressed this

**Comment by mtCarto** [comment]
Unable to change the items per page or show all in the project documents tables, the page is set to "1", but table retains the default number of items.

File: src/app/project/project-documents/project-documents.component.ts [hunk 2]
```diff
@@ -918,6 +921,8 @@ export class ProjectDocumentsComponent implements OnInit, OnDestroy {
     window.scrollTo(0, 0);
     this.loading = true;
 
+    const cc = this.tableParams2;
```
Comment (mtCarto) [inline src/app/project/project-documents/project-documents.component.ts]: Seems to be a typo? Assuming this should be **tableParams**? Compile error thrown here

Comment (vesalink) [reply src/app/project/project-documents/project-documents.component.ts]: oops, yup, just a debug thing, fixed...

**Comment by mtCarto** [comment]
Changing the project documents table's **number of items** is only changing the number of items for the 1st page of documents. The paginator doesn't update the number of pages based on the number of items previously selected and is stuck with the default 10. We would like to address this before merging this in. 

Leaving this PR open for reference for the next team member who picks this up.

**Comment by dhlevi** [comment]
Added commit: https://github.com/bcgov/eagle-admin/pull/469/commits/8f3d760b47b3dcb72b0512094b3231af8bb5ddd5

The pageSize values were not being reset on paging. This should solve the issues with show all, page size reseting.

**Pull request merged**